File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        left,right = 0, len(s) - 1


        while left < right:
            while left < right and not self.isAlpha(s[left]):
                left += 1
            while left < right and not self.isAlpha(s[right]):
                right -= 1
            if s[left].lower() != s[right].lower():
                return False
            left +=  1
            right -= 1
        return True

        
    def isAlpha(self, c):
            return (
                ord("0") <= ord(c) <= ord("9") or
                ord("a") <= ord(c) <= ord("z") or
                ord("A") <= ord(c) <= ord("Z")
            )

File: 0125-valid-palindrome/0125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */


isAlpha = (c) => {
    return (
        ('A' <= c && c <= 'Z') ||
        ('a' <= c && c <= 'z') ||
        ('0' <= c && c <= '9')
    );
}

var isPalindrome = function (s) {
    let left = 0;
    let right = s.length -1 ;

    while (left < right) {
        while (left < right && !isAlpha(s[left]))
            left += 1
        while (left < right && !isAlpha(s[right]))
            right -= 1
        if (s[left].toLowerCase() !== s[right].toLowerCase())
            return false
        left += 1;
        right -= 1;
    }
    return true;

};